test/integrations/deployers/ERC4626DeployerCoordinator.t.sol

```diff
@@ -0,0 +1,206 @@
+                    minFees: IHyperdrive.Fees({
+                        curve: 0.001e18,
+                        flat: 0.0001e18,
+                        governanceLP: 0.15e18,
+                        governanceZombie: 0.03e18
+                    }),
+                    maxFees: IHyperdrive.Fees({
+                        curve: 0.1e18,
+                        flat: 0.01e18,
+                        governanceLP: 0.15e18,
+                        governanceZombie: 0.03e18
+                    }),
+                    linkerFactory: address(0xdeadbeef),
+                    linkerCodeHash: bytes32(uint256(0xdeadbabe))
+                })
+            )
+        );
+
+        // Deploy the coordinator.
+        coordinator = new MockHyperdriveDeployerCoordinator(
+            factory,
+            address(new ERC4626HyperdriveCoreDeployer()),
+            address(new ERC4626Target0Deployer()),
+            address(new ERC4626Target1Deployer()),
+            address(new ERC4626Target2Deployer()),
+            address(new ERC4626Target3Deployer()),
+            address(new ERC4626Target4Deployer())
+        );
+
+        // Start a prank as the factory address. This is the default address
+        // that should be used for deploying Hyperdrive instances.
+        vm.stopPrank();
+        vm.startPrank(factory);
+    }
+
+    function test_initialize_success_asBase() external {
+        // Deploy all of the target instances.
+        for (uint256 i = 0; i < 5; i++) {
+            coordinator.deployTarget(
+                DEPLOYMENT_ID,
+                config,
+                new bytes(0),
+                i,
+                SALT
+            );
+        }
+
+        // Deploy a Hyperdrive instance.
+        IHyperdrive hyperdrive = IHyperdrive(
+            coordinator.deploy(DEPLOYMENT_ID, config, new bytes(0), SALT)
+        );
+
+        // Initialization should succeed with Alice as the initializer.
+        vm.stopPrank();
+        vm.startPrank(alice);
+        uint256 contribution = 100_000e18;
+        baseToken.mint(contribution);
+        baseToken.approve(address(coordinator), contribution);
+        vm.stopPrank();
+        vm.startPrank(factory);
+        uint256 lpShares = coordinator.initialize(
+            DEPLOYMENT_ID,
+            alice,
+            contribution,
+            0.05e18,
+            IHyperdrive.Options({
+                asBase: true,
+                destination: alice,
+                extraData: new bytes(0)
+            })
+        );
+
+        // Ensure the initializer received the correct amount of LP shares.
+        assertEq(
+            lpShares,
+            contribution.divDown(
+                hyperdrive.getPoolConfig().initialVaultSharePrice
+            ) - 2 * hyperdrive.getPoolConfig().minimumShareReserves
+        );
+        assertEq(lpShares, hyperdrive.balanceOf(AssetId._LP_ASSET_ID, alice));
+    }
+
+    function test_initialize_success_asShares() external {
```

Comment:
It would really be better if we did this in the contract we're inheriting from so that we get this test coverage across all of the deployment providers. I get that you probably don't want to spend the time doing that, but we might as well add a TODO so that we can keep track of the technical debt.

Reply:
the code doesn't work for steth otherwise I would have

Reply:
actually I was still working on getting this test to work with steth 😅